GANDLF/metrics/classification.py

```diff
@@ -25,6 +23,36 @@ def overall_stats(
         params["problem_type"] == "classification"
     ), "Only classification is supported for these stats"
 
+    def __convert_tensor_to_int(input_tensor: torch.Tensor) -> torch.Tensor:
+        """
+        Convert the input tensor to integer format.
+
+        Args:
+            input_tensor (torch.Tensor): The input tensor.
+
+        Returns:
+            torch.Tensor: The tensor converted to integer format.
+        """
+        return_tensor = input_tensor
```

Comment:
It needs to be checked but I believe such assignment still modifies input tensor inplace. I suppose if we want to avoid this side effect here deepcopy would be needed

Reply:
Fixed.